File: arrays/k_frequent/k_frequent.ts
export class Solution {
    /**
    * @param {string} testInput
    * @return {any} Returns either boolean to validate the test cases or a custom result. 
    */
    run_solution(testInput: any) {
        // nums = [1,2,2,3,3,3], k = 2
        const [nums, k] = testInput;
        const mapObj = {};
        const result = [];
        nums.forEach(element => {
            if(mapObj[String(element)]) {
                mapObj[String(element)]++
            } else {
                mapObj[String(element)] = 1
            }
            if(mapObj[String(element)] >= k) {
                result.push(element)
            }
        });
        return [...new Set(result)]
    }
}

File: arrays/k_frequent/k_frequent.py
""" Solution Module """


class Solution:
    """ Solution Class """

    def run_solution(self, test_input: list[list[int], int]):
        """ Solution Method """
        [nums, k] = test_input
        occurance_dict: dict[int, int] = dict()
        result = list()
        for i in nums:
            if i in occurance_dict:
                occurance_dict[i] += 1
            else:
                occurance_dict[i] = 1

            if occurance_dict[i] >= k:
                result.append(i)
        return list(set(result))
